### PR TITLE
add API to list performance profiles

### PR DESCRIPTION
**Pull request by khansaad**
This PR contains API changes to list the performance profiles present.

Signed-off-by: saakhan <saakhan@redhat.com>

### CODE REVIEW

File: src/main/java/com/autotune/utils/ServerContext.java [hunk 1]
```diff
@@ -37,6 +37,8 @@ public class ServerContext {
     public static final String UPDATE_RESULTS = ROOT_CONTEXT + "updateResults";
     public static final String RECOMMEND_RESULTS = ROOT_CONTEXT + "listRecommendations";
     public static final String CREATE_PERF_PROFILE = ROOT_CONTEXT + "createPerformanceProfile";
+    public static final String LIST_PERF_PROFILE = ROOT_CONTEXT + "listPerformanceProfile";
```
Comment (dinogun) [inline src/main/java/com/autotune/utils/ServerContext.java:40]: Should be a plural, `listPerformanceProfiles`

**Comment by dinogun** [comment]
Please add a `PerformanceProfiles.md` and move both create and list into that. Also the current md file for create is just a template. Please add the actual documentation

File: src/main/java/com/autotune/utils/ServerContext.java [hunk 2]
```diff
@@ -37,6 +37,8 @@ public class ServerContext {
     public static final String UPDATE_RESULTS = ROOT_CONTEXT + "updateResults";
     public static final String RECOMMEND_RESULTS = ROOT_CONTEXT + "listRecommendations";
     public static final String CREATE_PERF_PROFILE = ROOT_CONTEXT + "createPerformanceProfile";
+    public static final String LIST_PERF_PROFILE = ROOT_CONTEXT + "listPerformanceProfiles";
```
Comment (dinogun) [inline src/main/java/com/autotune/utils/ServerContext.java:40]: This needs to be `LIST_PERF_PROFILES` as well

File: design/PerformanceProfiles.md [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+}
+```
+#### Note: One of query or aggregation_functions is mandatory. Both can be present together.
+
+## Update Performance Profile
+
+Update Performance Profile data using input JSON as follows.
+
+**Request**
+`POST /updatePerformanceProfiles`
```
Comment (dinogun) [inline design/PerformanceProfiles.md:61]: Is this API supported in the code now?
Also this should be `updatePerformanceProfile` (singular)

File: design/PerformanceProfiles.md [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+            "hpoAlgoImpl": "optuna_tpe",
+            "function_variables": [
+                {
+                    "name": "cpuRequest",
+                    "query": "",
+                    "datasource": "prometheus",
+                    "value_type": "double",
+                    "kubernetes_object": "container",
+                    "trialSummaryResult": {},
+                    "cycleDataMap": {},
```
Comment (dinogun) [inline design/PerformanceProfiles.md:281]: These should not be part of the JSON

File: design/CreatePerformanceProfile.md [hunk 5]
```diff
@@ -5,30 +5,42 @@ tuned.
 
 # Attributes
 
-* name \
+- **name** \
```
Comment (dinogun) [inline design/CreatePerformanceProfile.md:8]: Rename this file to `PerformanceProfile.md`. Add a link to the API associated with PerformanceProfiles at the end and rename that file back to `PerformanceProfileAPI.md`

File: design/PerformanceProfiles.md (hunk 3 above)
Comment (khansaad) [reply design/PerformanceProfiles.md]: Removed it for now as this API is still in-progress.

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**